### PR TITLE
Handle complex return types

### PR DESCRIPTION
**Pull request by JonasVautherin**
Check if the return type is a primitive type or not. If not, the message/enum needs to be translated.

@xvzf

### CODE REVIEW

File: pb_plugins/dcsdkgen/utils.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ def extract_string_type(field):
     else:
         return "UNKNOWN_TYPE"
 
+def is_primitive_type(field):
+    """ Check if the field type is primitive (e.g. bool,
+    float) or not (e.g message, enum) """
+    return (not field.type in { 11, 14 })
```
Comment (xvzf) [inline pb_plugins/dcsdkgen/utils.py]: `(...)` is optional ;)

File: pb_plugins/templates/swift/method_single.j2 [hunk 2]
```diff
@@ -3,8 +3,14 @@ public func {{ name }}() -> Single<{{ return_type }}> {
...
-            let {{ name }}Response = try self.service.{{ name.lower() }}({{ name }}Request)
-            single(.success({{ name }}Response.{{ return_name }}))
+            let response = try self.service.{{ name.lower() }}({{ name }}Request)
+            {% if is_return_type_primitive -%}
+               let {{ return_name }} = response.{{ return_name }}
+            {% else -%}
+                let {{ return_name }} = {{ return_type }}.translateFromRPC(response.{{ return_name }})
+            {%- endif %}
+
+            single(.success({{ return_name }}))
```
Comment (xvzf) [inline pb_plugins/templates/swift/method_single.j2]: Nice!

File: pb_plugins/dcsdkgen/utils.py (hunk 1 above)
Comment (xvzf) [inline pb_plugins/dcsdkgen/utils.py]: How do you  get to 11 and 14?

Comment (JonasVautherin) [reply pb_plugins/dcsdkgen/utils.py]: https://github.com/grpc/grpc/blob/master/third_party/nanopb/generator/proto/google/protobuf/descriptor.proto#L128

**Pull request merged**